9417

simon2k opened the pull request:
Refactoring see spec.

simon2k committed:
9417

Refactoring "see" specs and dividing this into smaller specs:
  * texts spec
  * values spec
  * spcial selectors spec
  * scoping spec

Creating dummy files for specs:
  * texts.html
  * forms.html

simon2k committed:
9417

Refactoring see specs.

Creating dummy files for specs:
  * scopes.html
  * special_elements.html
  * elements_counter.html

simon2k committed:
9417

Refactoring see spec.

cs3b commented on spec/unit/see/form_elements/checkboxes_spec.rb:
```diff
@@ -0,0 +1,49 @@
+#encoding: utf-8
+require 'spec_helper'
+
+describe 'see form elements - checkboxes' do
```
might be good to group specs by contexts (checked / unchecked ) only the last spec would be within separate context

simon2k replied:
Okey

cs3b commented on spec/unit/see/form_elements/fields/disabled_spec.rb:
```diff
@@ -0,0 +1,32 @@
+
+  it 'should verify status for id' do
+    @user.see :disabled => 'disabledInput'
+    @user.not_see :disabled => 'xlInput'
```
little weird conditions it's enough to judge that there is no field on the page - and we really don't care what type this field is 

https://github.com/Skieloch/kameleon/blob/9417/lib/kameleon/dsl/see.rb#L93

then only thing that it would be nice to extent it is to able skip info about value, sometimes we don't care what value have field - we just doesn't want to have field with any value

```
 not_see :field => 'xlInput'
```

togethere with

```
 see :field => 'xlInput'
```

then above code should be extended to handle this special case, and get rid off 

```
 not_see :disabled
```

cs3b commented on spec/unit/see/form_elements/fields/empty_spec.rb:
```diff
@@ -0,0 +1,32 @@
+
+  it 'should verify status for id' do
+    @user.see :empty => 'prependedInput'
+    @user.not_see :empty => 'xlInput'
```
it should verify does field exists and have non empty value - where is this implemented?

cs3b commented on spec/unit/see/form_elements/fields/readonly_spec.rb:
```diff
@@ -0,0 +1,32 @@
+
+  it 'should verify status for id' do
+    @user.see :readonly => 'readonlyInput'
+    @user.not_see :readonly => 'xlInput'
```
as same as disabled we do need support this kind of structure - and I don't see if it's implemented anywhere ?

cs3b commented on spec/unit/see/form_elements/multiple_selects_spec.rb:
```diff
@@ -0,0 +1,45 @@
+require 'spec_helper'
+
+describe 'see form elements - multiple selects' do
```
group by context (selected / not selected) and add one mixed case at the end

cs3b commented:
what you think about splitting see / not_see in different context?

cs3b commented on spec/unit/see/form_elements/fields/readonly_spec.rb:
```diff
@@ -0,0 +1,32 @@
+require 'spec_helper'
+
+describe 'see readonly fields' do
```
within readonly it would be good to judge also value of the field if passed

```
see :readonly => ['Readonly input', 'Readonly textarea']
```

and

```
see :readonly => { 'some text' => 'Readonly input', 'some text inside' => 'Readonly textarea'}
```

cs3b commented on spec/unit/see/form_elements/radio_buttons_spec.rb:
```diff
@@ -0,0 +1,35 @@
+#encoding: utf-8
+require 'spec_helper'
+
+describe 'see form elements - radio buttons' do
```
context nest
checked
unchecked

and we don't need negation not_see 

user see checked / unchecked / or do not see field at all

cs3b commented on spec/unit/see/form_elements/selects_spec.rb:
```diff
@@ -0,0 +1,38 @@
+require 'spec_helper'
+
+describe 'see form elements - selects' do
```
group by context [:selected, :unselected]

cs3b commented on spec/unit/see/special_elements/error_message_for_spec.rb:
```diff
@@ -0,0 +1,24 @@
+  end
+
+  it 'should see many messages at once' do
+    @user.see :error_message_for => ['name', 'title', 'content']
```
it's relatively easy to implement plural version

```
see :error_messages_for => ['name', 'title', 'content']
```

cs3b commented on spec/unit/see/special_elements/links_spec.rb:
```diff
@@ -0,0 +1,25 @@
+require 'spec_helper'
+
+describe 'see special elements - links' do
```
could be intresting to be able to put links without src as array

```
 @user.see :links => ['What you want', 'What I need']
```

cs3b commented:
In future it would be nice to support two ways of passing text as matcher 
if string it should be strict matching 
if regex then regex matching

what you think?

simon2k committed:
Updated specs for checkboxes.

Grouped examples in context of checkboxes status.

simon2k committed:
Added specs fields_specs in order to check if fields exist.

simon2k committed:
Updated specs for check disabled fields.

Removed checking if field is disabled and instead of this was added checking if field exists.

simon2k committed:
Added case "empty" for not_see method in order to check if field exists and is has value.

simon2k committed:
Added case "readonly" for not_see method in order to check if field exist and is not readonly.

simon2k committed:
Added case "disabled" for not_see method in order to check if field exist and is enabled.

simon2k committed:
Refactoring #see and #not_see methods in See module.

simon2k committed:
Updated specs "see multiple selects".

Grouped examples in context of selects status.

simon2k committed:
Extended verifications for read only fields:
  * checking if field has or not has value.

simon2k committed:
Removal of verification for checking whether if a field is disabled.

simon2k committed:
Removal of checking if user cannot see checked or unchecked fields from specs:
  * checkboxes
  * radio buttons

simon2k committed:
Updated specs for selects.

Grouped examples in context of selects status.

simon2k committed:
Updated specs for radio buttons.

Grouped examples in contexts of radio buttons status.

simon2k committed:
Added plurar version for see "error_message_for" - "error_messages_for"

simon2k committed:
Added in "see" method checking if exist links without supplied src paths.

simon2k commented:
> what you think about splitting see / not_see in different context?

It may be good.. :-)

simon2k committed:
Refactoring module See - using internal matchers instead of Capybara matchers.

cs3b commented:
yes - you can do it

On 7 January 2012 11:10, Szymon Kieloch
reply@reply.github.com
wrote:

> > what you think about splitting see / not_see in different context?
> 
> Spliting in two context "see" & "not_see" hm it may be good.. :-)
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/cs3b/kameleon/pull/1#issuecomment-3395063

## 

Michał Czyż - Team Leader @ selleo.com
mobile: +48 661 120 232 | skype: mc_selleo
linkd.in/michalczyz | twitter.com/#!/cs3b

simon2k committed:
Removed unnecessary spec fiele.

simon2k committed:
Splitting specs by context of #see and #not_see methods.

simon2k committed:
Added missing cases in the method #not_see.

simon2k committed:
Removed unnecessary exmaple in the spec: "#not_see special elements - links".

cs3b commented:
you need to rebase you branch over master - it's not straight forward can not  perform merge
